concurrency/runner.go

```diff
@@ -60,47 +61,37 @@ func ForEachUser(ctx context.Context, userIDs []string, concurrency int, userFun
 	return errs.Err()
 }
 
-// ForEach runs the provided jobFunc for each job up to concurrency concurrent workers.
+// ForEachJob runs the provided jobFunc for each job index in [0, jobs) up to concurrency concurrent workers.
 // The execution breaks on first error encountered.
-func ForEach(ctx context.Context, jobs []interface{}, concurrency int, jobFunc func(ctx context.Context, job interface{}) error) error {
-	if len(jobs) == 0 {
+func ForEachJob(ctx context.Context, jobs int, concurrency int, jobFunc func(ctx context.Context, idx int) error) error {
```

Comment:
I think we should keep the old function and perhaps deprecate it or reimplement on top of new function. We're learning that breaking existing API too much is too disruptive for our projects, due to interdependencies between them.

Reply:
Fair enough, I brought it back in https://github.com/grafana/dskit/pull/113/commits/6faae529a2161e30c589b64dcd54f1d1ad6e8575 with proper `// Deprecated.` comment so IDEs would bring users' attention.